dev-libs/openssl/openssl11-1.1.0f.recipe

```diff
@@ -84,18 +115,33 @@ INSTALL()
 	prepareInstalledDevelLibs libcrypto libssl
 	fixPkgconfig
 
+	if [ -n "$secondaryArchSuffix" ]; then
+		maybe_docDir_html_man3=
+		maybe_manDir_man3=
+	else
+		maybe_docDir_html_man3=$docDir/html/man3
+		maybe_manDir_man3=$manDir/man3
+	fi
+
 	# devel package
 	packageEntries devel \
 		$developDir \
-		$docDir \
-		$manDir/man3
+		$maybe_docDir_html_man3 \
+		$maybe_manDir_man3
 
-	# Remove stuff we don't need in the secondary architecture base package,
-	# since we make it depend on the primary package.
+	# Remove stuff we don't need in the secondary architecture base package
 	if [ -n "$secondaryArchSuffix" ]; then
 		rm -rf $prefix/bin
 		rm -rf $dataRootDir/ssl
 		rm -rf $documentationDir
+	else
+		if [ -n "$cmdSuffix" ]; then
+			mv $binDir/openssl $binDir/openssl-$libVersion
+			mv $binDir/c_rehash $binDir/c_rehash-$libVersion
```

Comment:
Should be: mv $binDir/c_rehash $binDir/c_rehash$cmdSuffix or something with -
Instead of another subpackage, please also rename the man pages and package them directly in the package containing the commands.

Reply:
Renaming the man pages will be tricky, but I'll try.
Here's a small update to fix the first issue you pointed.

Reply:
I fear that renaming the man pages will be a nightmare :-(
For 1.0.2 there are currently 100 files in documentation/man/man1.
Among these 100 files we have:
* openssl.1
* 42 files which start with openssl-
* 56 files which do not start with openssl

To keep the man pages in the base package, we could create a 1ssl section for the man pages of 1.0.2 and a 1ssl11 section for those from 1.1.x.
But having extra man sections might also be quite confusing.

Given that openssl 1.1.x has its html doc in the base package, would you agree with keeping the man pages in their sub-package ?
Most users who might be interested in reading openssl's man pages will probably know how to install openssl_man or openssl11_man.
Hmm, maybe we should add a "RECOMMENDS" keyword to haikuporter ;-)
Unless HaikuDepot already makes use of the SUPPLEMENTS (which I have already added.)

Hmm, do you think that "moving" (and renaming, for 1.1.x) to the base package *only* the 43 files (which start with openssl) would be a good idea? Oops, it will also be confusing...

Reply:
The general policy is that it is not possible to install _dev version of packages for two different versions at the same time (there can be exceptions). I would say the same could apply to the manpages.

We could even provide only the most up to date version of the docs, and do not package docs at all for older OpenSSL versions. I think their documentation do mentions the version when a feature was introduced, so this should not be a problem even for people still using the old version (they can even see what they are missing).

Reply:
Yep, I (already) declared the CONFLICTS for both _devel and _man sub-packages, so we are safe here.
1.0.2 does not provide html docs. (I tried to have it produce these docs, but I suspect that it was not easily available on 1.0.2.)
I think the current commit is the best compromise we can have.

BTW, I'm just added `cmd:openssl{,-1.1}` to the debuginfo package.

Reply:
> We could even provide only the most up to date version of the docs, and do not package docs at all for older OpenSSL versions.

If we choose to only ship the docs from 1.1.x, this means we cannot put it add it to the base package of 1.0.2 (unless we add the 1.1.x tarball to the 1.0.2 recipe, but that would not be a good idea.)
So we would need to ship the 1.1.x docs outside of 1.0.2.
Now, if we put the docs for 1.1.x in the base package for 1.1.x, then anyone willing to have the runtimes for 1.0.2 would be forced to install the 1.1.x base package.

So the best split is to have the 1.1.x doc in a _man sub-package for 1.1.x.
And since 1.1.x is still marked as untested, we also need to ship the docs for 1.0.2. The best place for that is in a _man sub-package for 1.0.2.

The current commit in this PR does exactly this ;-)

BTW, the tests work fine for 1.0.2. But they are not working for 1.1.x. When I launch the tests for 1.1.0 it does a few things than it hangs, doing nothing. There is probably a process waiting (forever) for standard input. I did not dig it.
BTW, we'll soon have mbedtls, which passes all tests :tada:

Reply:
I would rename the man files anyway, while updating the links between pages. Not a big deal for a sed expert :)
For instance openssl.1 => openssl_11.1, then update "L<openssl(1)>" => "L<openssl_11(1)>"

Reply:
We could to that, but what would you do with the 56 files which do not even contain "openssl" in their names?
I think it will be much easier for the users to install the _man sub-package they want, and then use the same names as in all other OSes.
Those users who would be lost with the installion of the _man sub-package would not even use the man command. They would use Google ;-)

Reply:
Shall we leave the man pages in their own sub-packages, without worrying about renaming any man page?
Reminder: WIth 1.1.x the html docs are *already* in the base package ;-)